### ISSUE TITLE
Debugger step out doesn't work

### ISSUE THREAD

**Issue by ericwinger** [ericwinger/Jade#79]
- Using the reproduction case in #75, step until you're inside the block. 
- Then press the Step Out button. 

I believe that this should take you out of the block, but instead just takes you to the next step point in the block.

**Comment by LisaAlmarode** [comment]
Step out is still apparently completely broken (i.e., just does a step) in 3.0.75.

**Comment by ericwinger** [comment]
Yes, it appears to effectively have executed stepOver instead of stepOut. I implemented a fix in 3.0.76 like this. 

1. If you have the top frame selected, stepOut will step to the next highest frame (frame 2 in the stack).
2. If you have a different frame selected, it will stepOut to that frame. 

This seems (to me) to be the most predictable and useful behavior for the user.

**Comment by LisaAlmarode** [comment]
It's much more out-stepish in 3.0.76.  Will leave it to someone who uses the debugger heavily to determine the correct behavior and if there are deviations therefrom.
